diesel/src/deserialize.rs

```diff
@@ -578,3 +578,25 @@ where
 {
     const FIELD_COUNT: usize = <ST as crate::util::TupleSize>::SIZE;
 }
+
+/// A helper trait for giving a type a useful default value.
+///
+/// This is needed for types that can be used as range to represent the empty range as
+/// (Bound::Excluded(DEFAULT), Bound::Excluded(DEFAULT)).
+/// When possible, implementations of this trait should fall back to using std::default::Default.
+pub trait Defaultable {
```

Comment:
We do not want to make this a public trait for now, as this is just a workaround. I would rather prefer to go the way to keep this trait private and have the wild card impl below instead. In that way other crates can still support it by implementing `Default`.

Reply:
The wild card is what I tried first but couldn't get it working due to the conflicting implementations error, e.g. `"conflicting implementations of trait MyDefault for type OffsetDateTime ... upstream crates may add a new impl of trait std::default::Default for type OffsetDateTime in future versions"`. But indeed long term that would be nicer.

Is it ok if I just mark the `Defaultable` trait as `pub(crate)`? Or do you have another idea in mind?

Reply:
It's fine to just use `pub(crate)` then. I read the comment as: This is not possible because it will cause errors in crates depending on diesel and not as this is not possible because in causes errors in diesel itself.

Reply:
Sorry for the confusion. Reading it again it's indeed a bit misleading. I just copied it from here: https://github.com/diesel-rs/diesel/blob/a3ea5242567b1a44f3b606d9b552bdd23065e001/diesel/src/deserialize.rs#L553

I updated the MR to have the trait private to the crate.

Reply:
Thanks, looks good now